alts/src/main/java/io/grpc/alts/internal/AltsProtocolNegotiator.java

```diff
@@ -20,14 +20,11 @@
 
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.collect.ImmutableList;
-import com.google.protobuf.Any;
 import io.grpc.Attributes;
 import io.grpc.Channel;
-import io.grpc.Grpc;
-import io.grpc.InternalChannelz.OtherSecurity;
-import io.grpc.InternalChannelz.Security;
 import io.grpc.SecurityLevel;
 import io.grpc.Status;
+import io.grpc.alts.AltsAuthContext;
```

Comment:
This creates a circular dependency between the alts and alts.internal packages. @jiangtaoli2016, I think this needs more design to figure out how the components will land, especially when tsi is split out into its own package.

Reply:
Another way to approach this, is to use the existing AltsContext/AltsAuthContext split. That means the AltsContext is used within the alts code and AltsAuthContext outside. For this the AltsContext is newly stored in attributes and there is a non-internal static-factory method to create a AltsAuthContext given a ServerCall (or attributes). This way the internal attribute constant doesn't need to be exposed.
Note that this would also solve your comment about the toChannelSecurity method.

Reply:
Martin: what info do you need from AltsAuthContext?
How about keep AltsAuthContext untouched (still remain in internal). We create wrapper in Alts to plumb the attribute you need -- this would be similar to `alts/AuthorizationUtil.java`

Reply:
It looks like you only need protocol, security_level, and user name. We can create a wrapper class and provide in the alts package.

Reply:
I only need the `peerServiceAccount` and the security level for now. Given that the security level is fixed to `INTEGRITY_AND_PRIVACY` in `AltsAuthContext` - I could also live without it for now (although I don't like to duplicate that assumption into my code).  Therefore, I propose to go forward in three steps/PR and close this one: 
1.  I create a new PR to introduce the wrapper and the plumbing to create the wrapper from a `ServerCall`
2. We make the security level accessible by changing its proto package (or introduce a mapping to a non-internal security level)
3. Introduce the security level in the wrapper (from step 1) 
WDYT?

Reply:
I don't understand (1). (2) sounds like it is making all the protos non-internal; it seems better to just make a copy of the enum. Unclear how (3) would be plumbed.

Reply:
After I careful looked at the code, the main problem here is that in `AltsAuthContext`, there is a method `getPeerRpcVersions()` which is needed by AltsProtocolNegotiator (which is alts.internal) for validation purpose. That is the root cause of cyclic dependency.

There are two options in my view
1. Create a visibility controlled function in alts package, something like `getAltsAuthContext(ServerCall<?, ?> call)`, which will return a stripped version of AltsAuthContext on whatever needed by @martin-schaub .
2. Create another class AltsSimpleContext in alts. Keep AltsAuthContext in alts.internal. Create different Attributes.key for AltsSimpleContext. During `fireProtocolNegotiationEvent`, set both keys. 

Or we just make one-off exception and allow @martin-schaub code to directly access alts.internal? 

Any suggestion @ejona86 ?

Reply:
Per offline discussion with @ejona86 , here is the recommendation.
1. Keep Rename AltsAuthContext to AltsInternalContext. Keep this class in alts.internal. Keep the same Attributes.key.
2. Create a wrapper class AltsContext in alts package. This is a facade of AltsInternalContext and delegate the necessary functions e.g., `getPeerServiceAccount`
3. Create a static utility class in alts package `AltsContext getAltsContext(ServerCall<?, ?> call)`, so that anyone can access AltsContext via this utility function.

Reply:
Thank you for your feedback. Implemented in 
* https://github.com/grpc/grpc-java/pull/7857
* https://github.com/grpc/grpc-java/pull/7862

I will close this PR.